File: 27_RemoveElement.cpp
#include<vector>
#include<algorithm>
#include<iostream>
#include<string>
#include<list>
using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        
        erase(nums, val);

        int n = nums.size();
        return n;
    }
};

File: 27_Remove Element.java
class Solution {

    public int removeElement(int[] nums, int val) {
        if(nums.length == 0)
        {
            return 0;
        }
        
        int i = 0;
        int n = nums.length - 1;

        while (i <= n) {
            if (nums[i] == val) {
                nums[i] = nums[n];
                n--;
            } else {
                i++;
            }
        }

        return i;
    }
}